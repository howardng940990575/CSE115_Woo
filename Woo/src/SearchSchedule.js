import React, { Component, ReactNode } from 'react';
import { StyleSheet, Text, View, Button, TextInput } from 'react-native';
import * as firebase from "firebase";
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
import { Fumi } from 'react-native-textinput-effects';
import { initialEmail } from './Loading.js';
import { ScrollView } from 'react-native-gesture-handler';

export default class SearchSchedule extends Component {

    constructor(props) {
        super(props);
        this.state = {
            events: "",
            email: "",
        };
        this.onQuery = this.onQuery.bind(this);
        this.searchEmail = this.searchEmail.bind(this);
    }

    onPressRequests = () => {
      this.props.navigation.navigate('PatientHomepage')
    }

    onQuery = (email) => {
        console.log('Searching for this user ' +email );
          this.searchEmail(email).then((res) => {
          this.setState({
            events: res,
          })
          console.log("Inquiry appointments: " + res);
        })
      }
  
      async searchEmail(email) {
         events = "\n";
         querySnapshot = await firebase.firestore().collection("users").doc(email).collection("events").get(); //.then(function(querySnapshot) {
         querySnapshot.forEach(function(doc) {
            console.log(doc.data())
            events += "----------------------------------------\nAppointment date " + doc.data().date +"\nDescription: " 
            + doc.data().description + "\nDoctor: " + doc.data().doctor + "\nHospital: " + doc.data().hospital + "\nat time: " + doc.data().time;
          });
          if (events === "\n" ) {
              events += "NO APPOINTMENTS FOUND OR INVALID EMAIL";
          }
          return events
  }

  render() {
    const { events } = this.state;
    const newString = events ? events.toString() : ''
      return (
        <ScrollView style = {styles.scrollView}>
          <View style = {styles.container}>
              <Text
                style = {styles.titleText}>Search for a users appointments</Text>
                <Fumi
                  label={'Email Address'}
                  iconClass={FontAwesomeIcon}
                  iconName={'pencil'}
                  iconColor={'white'}
                  inputStyle={{ color: '#000000' }}
                  onChangeText={email => { this.setState({email}) } }
                /> 
              <Button
                style={styles.buttonStyle}
                onPress = {() => this.onQuery(this.state.email)}
                title="Get specfic user appointments"
              />
              <Button
                onPress = {() => this.onPressRequests()}
                title="Go back"
              />
              <Text style = {styles.titleText}> Appointments found for user:</Text>
              <Text>{newString}</Text>
          </View>
        </ScrollView>
        );
    }
}


const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#72C3C9',
      borderTopWidth: 15,
      borderColor: '#72C3C9',
    },
    titleText: {
      paddingBottom: 16,
      borderColor: '#72C3C9',
      backgroundColor: '#72C3C9',
      textAlign: 'center',
      color: '#404d5b',
      fontSize: 20,
      fontWeight: 'bold',
      opacity: 0.8,
    },
    form : {
      justifyContent: 'center',
      marginTop: 50,
      padding: 20,
      backgroundColor: '#ffffff',
    },
    buttonStyle: {
      backgroundColor: '#fff',
      borderWidth: 1,
      borderColor: '#336633',
      paddingTop: 4,
      paddingBottom: 4,
      paddingRight: 25,
      paddingLeft: 25,
      marginTop: 10,
      width: 300
    },
    scrollView: {
      backgroundColor: '#72C3C9'
    }
  });