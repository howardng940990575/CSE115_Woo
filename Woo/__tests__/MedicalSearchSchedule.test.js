import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SearchSchedule from '../src/MedicalSearchSchedule';

test('Medical Search Schedule SnapShot', ()=> {
    const snap = renderer.create(<SearchSchedule />).toJSON();
    expect(snap).toMatchSnapshot();
});
