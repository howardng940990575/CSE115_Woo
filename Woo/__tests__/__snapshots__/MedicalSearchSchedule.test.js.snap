// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Medical Search Schedule SnapShot 1`] = `
<RCTScrollView
  collapsable={false}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  style={
    Object {
      "backgroundColor": "#72C3C9",
      "marginTop": 25,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderTopWidth": 15,
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#ffffff",
            "color": "#404d5b",
            "fontSize": 20,
            "fontWeight": "bold",
            "opacity": 0.8,
            "paddingBottom": 16,
            "textAlign": "center",
          }
        }
      >
        Search for a users appointments
      </Text>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "overflow": "hidden",
              "paddingTop": 16,
            },
            Object {
              "borderBottomWidth": 0.5,
              "borderColor": "rgba(0,0,0,0.2)",
              "borderTopWidth": 0.5,
            },
            Object {
              "height": 64,
            },
          ]
        }
      >
        <Text />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#f0f0f0",
                "position": "absolute",
                "top": 8,
                "width": 1,
              },
              Object {
                "height": 48,
                "left": 48,
              },
            ]
          }
        />
        <View
          accessible={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 48,
              "left": 56,
              "position": "absolute",
              "top": 24,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgba(105, 105, 105, 1)",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Email Address
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          animationDuration={300}
          autoCapitalize="none"
          height={48}
          iconClass={[Function]}
          iconColor="white"
          iconName="envelope-square"
          iconSize={20}
          iconWidth={40}
          inputPadding={16}
          inputStyle={
            Object {
              "color": "#000000",
            }
          }
          label="Email Address"
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          passiveIconColor="#a3a3a3"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "padding": 7,
                "paddingLeft": 0,
              },
              Object {
                "color": "white",
                "marginLeft": 56,
              },
              Object {
                "color": "#000000",
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 10,
              "marginHorizontal": 20,
              "marginVertical": 5,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.2,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#404d5b",
                "fontSize": 20,
              }
            }
          >
            Get specfic user appointments
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 10,
              "marginHorizontal": 20,
              "marginVertical": 5,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.2,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#404d5b",
                "fontSize": 20,
              }
            }
          >
            Go Back
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#ffffff",
            "color": "#404d5b",
            "fontSize": 20,
            "fontWeight": "bold",
            "opacity": 0.8,
            "paddingBottom": 16,
            "textAlign": "center",
          }
        }
      >
         Appointments found for user:
      </Text>
      <View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
              false,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "borderRadius": 10,
              },
              Object {
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 10,
              },
              undefined,
              undefined,
              Object {
                "backgroundColor": "#f6f5f5",
              },
              Object {
                "paddingBottom": 45,
                "paddingTop": 30,
                "zIndex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
                false,
                Object {
                  "flexDirection": "row",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "paddingHorizontal": 30,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <Text>
              NO APPOINTMENTS FOUND FOR THIS USER
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
